solution/backend/content_search/views.py

```diff
@@ -95,3 +102,73 @@ def list(self, request):
             return self.get_paginated_response(serializer.data)
         else:
             return Response(serializer.data)
+
+
+class PostContentTextViewset(viewsets.ViewSet):
+    # authentication_classes = [SettingsAuthentication]
+    # permission_classes = [IsAuthenticatedOrReadOnly]
+    @extend_schema(
+        description="Retrieve list sof uploaded files",
+        request=ContentUpdateSerializer,
+        responses={200: ContentUpdateSerializer}
+    )
+    def update(self, request, *args, **kwargs):
+        post_data = request.data
+        id = post_data['id']
+        text = post_data['text']
+        index = ContentIndex.objects.get(uid=id)
+        index.content = text
+        index.save()
+        return Response(data=f'Index was updated for {index.doc_name_string}')
+
+
+class InvokeTextExtractorViewset(viewsets.ViewSet):
+    @extend_schema(
+        description="Post to the lambda function",
+        request=InvokeTextExtractorSerializer,
+        responses={200: InvokeTextExtractorSerializer}
+    )
+    def extract(self, request, *args, **kwargs):
+        post_data = request.data
+        uid = post_data['uid']
+        post_url = post_data['post_url']
+        index = ContentIndex.objects.get(uid=uid)
+        if not index:
+            return Response({'message': "Index does not exist"})
+        json_object = {
+            'id': uid,
+            'uri': index.url,
+            'post_url': post_url,
+        }
+        if index.file:
+            try:
+                json_object['uri'] = index.file.get_key()
+                json_object['backend'] = 's3'
+                json_object["s3"] = {
+                                        "aws_access_key_id": settings.AWS_ACCESS_KEY_ID,
+                                        "aws_secret_access_key": settings.AWS_SECRET_ACCESS_KEY,
+                                        "aws_storage_bucket_name": settings.AWS_STORAGE_BUCKET_NAME
+                                    }
+            except ValueError:
+                json_object['backend'] = 'web'
+
+        if settings.USE_LOCAL_TEXTRACT:
+            json_object['textract'] = {
+                'aws_access_key_id': settings.TEXTRACT_KEY_ID,
+                'aws_secret_access_key': settings.TEXTRACT_SECRET_KEY,
+                'aws_region': 'us-east-1'
+            }
+            # return Response(json_object)
+            docker_url = 'http://host.docker.internal:8001/'
+            resp = requests.post(
+                docker_url,
+                data=json.dumps(json_object),
+                timeout=60,
+            )
+            resp.raise_for_status()
+        else:
+            lambda_client = build_lambda_client()
+            resp = lambda_client.invoke(FunctionName=settings.TEXTRACT_ARN,
```

Comment:
On local development this will attach to the  main DEV lambda to extract text, on the environment it will use the experimental one if one was built with it.